File: JavaScript/isAnagram.js
function isAnagram(s, t) {
  if (s.length !== t.length) {
    return false;
  }
  let countS = {};
  let countT = {};

  for (let i = 0; i < s.length; i++) {
    countS[s[i]] = (countS[s[i]] || 0) + 1;
    countT[t[i]] = (countT[t[i]] || 0) + 1;
  }
  for (let c in countS) {
    if (countS[c] !== countT[c]) {
      return false;
    }
  }
  return true;
}

console.log(isAnagram("car", "par"));

File: Python/isAnagram.py
def isAnagram(s, t):
    if len(s) != len(t):
        return False
    countS = {}
    countT = {}

    for i in range(len(s)):
        countS[s[i]] = countS.get(s[i], 0) + 1
        countT[t[i]] = countT.get(t[i], 0) + 1

    for c in countS:
        if countS[c] != countT.get(c, 0):
            return False
    return True


print(isAnagram("car", "car"))
